lib/modules/manager/gomod/update.ts

```diff
@@ -17,14 +17,15 @@ export function updateDependency({
 }: UpdateDependencyConfig): string | null {
   try {
     logger.debug(`gomod.updateDependency: ${upgrade.newValue}`);
-    const { depType, updateType } = upgrade;
+    const { depType } = upgrade;
     const currentName = upgrade.depName;
-    if (updateType === 'replacement') {
-      logger.warn('gomod manager does not support replacement updates yet');
-      return null;
+    let newName = currentName;
+    // newName will be available for replacement
+    if (upgrade.newName) {
+      newName = upgrade.newName;
     }
     // istanbul ignore if: should never happen
-    if (!currentName || !upgrade.managerData) {
+    if (!currentName || !newName || !upgrade.managerData) {
```

Comment:
```suggestion
    if (!currentName || !upgrade.managerData) {
```

Impossible to happen